add ability to do environment variables substitution (#14)
adevinta/zoe: docs/configuration/chain.md

The final zoe configuration will be the result of merging all the above configuration values in the increasing level priority.

The special `common.yml` configuration file is supposed to contain configuration values that are common to all the environments. Registered expressions or secrets providers are a good use case. 

## Environment variables substitutions

> It is possible to use environment variables inside the zoe yaml or json configuration files and zoe will lookup those variables at runtime from the environment and does an automatic substitution.

Reviewer: ```suggestion
It is possible to use environment variables inside the zoe yaml or json configuration files and zoe will automatically substitute those variables at runtime from the environment.
```
